homeassistant/components/escea/climate.py

```diff
@@ -0,0 +1,299 @@
+"""Support for the Escea HVAC."""
+from __future__ import annotations
+
+from collections.abc import Callable, Coroutine
+import logging
+from typing import Any
+
+from pescea import Controller
+
+from homeassistant.components.climate import ClimateEntity
+from homeassistant.components.climate.const import (
+    FAN_AUTO,
+    FAN_HIGH,
+    FAN_LOW,
+    HVAC_MODE_HEAT,
+    HVAC_MODE_OFF,
+    SUPPORT_FAN_MODE,
+    SUPPORT_TARGET_TEMPERATURE,
+)
+from homeassistant.const import (
+    ATTR_TEMPERATURE,
+    CONF_EXCLUDE,
+    PRECISION_WHOLE,
+    TEMP_CELSIUS,
+)
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.typing import ConfigType
+
+from .const import (
+    DATA_CONFIG,
+    DATA_DISCOVERY_SERVICE,
+    DISPATCH_CONTROLLER_DISCONNECTED,
+    DISPATCH_CONTROLLER_DISCOVERED,
+    DISPATCH_CONTROLLER_RECONNECTED,
+    DISPATCH_CONTROLLER_UPDATE,
+    ESCEA,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+_ESCEA_FAN_TO_HA = {
+    Controller.Fan.FLAME_EFFECT: FAN_LOW,
+    Controller.Fan.FAN_BOOST: FAN_HIGH,
+    Controller.Fan.AUTO: FAN_AUTO,
+}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, config: ConfigType, async_add_entities: Callable
+):
+    """Initialize an Escea Controller."""
+    disco = hass.data[DATA_DISCOVERY_SERVICE]
+
+    @callback
+    def init_controller(ctrl: Controller):
+        """Register the controller device."""
+        conf: ConfigType = hass.data.get(DATA_CONFIG)
+
+        # Filter out any entities excluded in the config file
+        if conf and ctrl.device_uid in conf[CONF_EXCLUDE]:
+            _LOGGER.info("Controller UID=%s ignored as excluded", ctrl.device_uid)
+            return
+        _LOGGER.info("Controller UID=%s discovered", ctrl.device_uid)
+
+        device = ControllerDevice(ctrl)
+        async_add_entities([device])
+
+    # create any components not yet created
+    for controller in disco.pi_disco.controllers.values():
+        init_controller(controller)
+
+    # connect to register any further components
+    async_dispatcher_connect(hass, DISPATCH_CONTROLLER_DISCOVERED, init_controller)
+
+    return True
+
+
+def _return_on_connection_error(ret: Any = None):
+    def wrap(func: Callable):
+        def wrapped_f(*args, **kwargs):
+            if not args[0].available:
+                return ret
+            try:
+                return func(*args, **kwargs)
+            except ConnectionError:
+                return ret
+
+        return wrapped_f
+
+    return wrap
+
+
+class ControllerDevice(ClimateEntity):
+    """Representation of Escea Controller."""
+
+    def __init__(self, controller: Controller) -> None:
+        """Initialise ControllerDevice."""
+        self._controller = controller
+
+        self._supported_features = SUPPORT_FAN_MODE
+        self._supported_features |= SUPPORT_TARGET_TEMPERATURE
```

Comment:
```suggestion
        self._attr_supported_features = SUPPORT_FAN_MODE | SUPPORT_TARGET_TEMPERATURE
```
This allows removing the `supported_features` property getter from below.